deployments/k8s/default/discovery-engine/clusterrole.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: discovery-engine-role
+rules:
+- apiGroups: ["*"]
+  resources: ["pods", "services", "deployments", "endpoints", "namespaces", "nodes","replicasets", "statefulsets", "daemonsets", "secrets"]
+  verbs: ["get", "list", "watch","create", "update", "delete"]
```

Comment:
Why do we need create/delete permissions for nodes? I think the Clusterroles can be refined.

Reply:
i used existing clusterrole for discovery-engine here, ig we don't require create/delete permission for most of the resources, can you validate it? for what resources we can remove create/delete permissions?

Reply:
@Ankurk99 @rksharma95 , can we have this in a seperate issue ?